allennlp/interpret/attackers/input_reduction.py

```diff
@@ -0,0 +1,144 @@
+from typing import List
+import numpy as np
+import torch
+from allennlp.interpret.attackers import Attacker
+from allennlp.common.util import JsonDict, sanitize
+from allennlp.data import Instance
+
+@Attacker.register('input-reduction')
+class InputReduction(Attacker):
+    """
+    Runs the input reduction method from `Pathologies of Neural Models Make Interpretations
+    Difficult` https://arxiv.org/abs/1804.07781, which removes as many words as possible
+    from the input _without_ changing the model's prediction.
+    """
+    def attack_from_json(self, inputs: JsonDict,
+                         name_of_input_field_to_attack: str = 'tokens',
+                         name_of_grad_input_field: str = 'grad_input_1',
+                         ignore_tokens: List[str] = ["@@NULL@@"]): # pylint: disable=dangerous-default-value
+        original_instances = self.predictor.inputs_to_labeled_instances(inputs)
+        final_tokens = []
+        fields_to_check = {}
+        for current_instances in original_instances:
+            current_instances = [current_instances]
```

Comment:
Same comment about not putting this in the list.  Also, it'd be nice if you used the same name in both places.  Maybe just `for instance in original_instances`?  It doesn't look like there's a previous `instance` variable that you would be overriding.